src/dataStructures.hpp

```diff
@@ -205,33 +205,158 @@ struct SpongeZoneData {
   bool singleTemperature;
 };
 
-struct volumeFaceIntegrationArrays {
-  // nodes IDs and indirection array
-  Array<int> nodesIDs;
-  Array<int> posDofIds;
-  // count of number of elements of each type
+/** @brief Array providing information about dof indices for elements
+ *
+ * These arrays store information about the degrees of freedom
+ * associated with each element.  They are used in the _GPU_ code path
+ * to provide information that would otherwise (i.e., on the cpu path)
+ * come from, for instance, a FiniteElement or FiniteElementSpace
+ * object.
+ */
+struct elementIndexingData {
+  /** Dof array offset for element i
+   *
+   *  To be used with element_dofs_list.  Specifically,
+   *
+   *  element_dofs_list[element_dof_offset[i] + j] = index of jth dof of element i
+   */
+  Array<int> element_dof_offset;
+
+  /** Number of dofs for each element
+   *
+   *  element_dof_number[i] = number of dofs for element i
+   */
+  Array<int> element_dof_number;
+
+  /** List of dof indices, ordered by element.
+   *
+   *  I.e. [ (dof indices for elem 0), (dof indices for elem 1), ... (dof indices for elem Ne-1) ]
+   *
+   *  Since elements may have different numbers of dofs, use
+   *  element_dof_offset to aid in indexing as follows:
+   *
+   *  element_dofs_list[element_dof_offset[i] + j] = index of jth dof of element i
+   */
+  Array<int> element_dofs_list;
+
+  /** Number of elements of each type (e.g, number of tets, number of hexes, etc) */
   Array<int> numElems;
-  // Array<int> posDofQshape1; // position, num. dof and integration points for each face
-  Vector shapeWnor1;  // shape functions, weight and normal for each face at ach integration point
-  Vector shape2;
+};
 
+/** @brief Storage for data used in the _GPU_ code path
```

Comment:
This comment is stale.  Update it to be specific to `interiorFaceIntegrationData`.

Reply:
788efb7